hbase_sql/src/main/java/com/splicemachine/derby/impl/sql/ZkPropertyManager.java

```diff
@@ -67,8 +67,14 @@ public String getProperty(String propertyName) throws StandardException{
 
     @Override
     public void addProperty(String propertyName,String propertyValue) throws StandardException{
+        String path = zkSpliceDerbyPropertyPath + "/" + propertyName;
         try{
-            ZkUtils.safeCreate(zkSpliceDerbyPropertyPath+"/"+propertyName,Bytes.toBytes(propertyValue),ZooDefs.Ids.OPEN_ACL_UNSAFE,CreateMode.PERSISTENT);
+            if (ZkUtils.getRecoverableZooKeeper().exists(path, false) == null) {
+                ZkUtils.safeCreate(path, Bytes.toBytes(propertyValue), ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT);
+            }
+            else {
+                ZkUtils.setData(path, Bytes.toBytes(propertyValue), -1);
+            }
```

Comment:
This is not an atomic operation anymore, for a good reason, as `safeCreate` does [not](https://github.com/sfines/menagerie/blob/master/src/main/java/org/menagerie/ZkUtils.java#L206) alter the data if the path exist.

Does it make sense to make it atomic, e.g. by obtaining a distributed lock on the `path` itself?

By the way, is setting a property a transactional operation? at first glance it seems not, right, from the looks of this method, it looks like last-writer-wins, causing this anomaly:

```
client 1                                                       client 2

autocommit off                                                 autocommit off
call syscs_util.syscs_set_global_database_property('a','b');
call syscs_util.syscs_set_global_database_property('c','e');
                                                               call syscs_util.syscs_set_global_database_property('a','z');
commit;
                                                               commit;
```

End result:
```
PropertyName,PropertyValue
a,z
c,e
```

Reply:
property is not transactional

Reply:
If the path exists, the data is modified.

Reply:
> Filed DB-12091 to followup Youssef's comments

Thank you @jyuanca